### PR TITLE
Use controller-gen to generate the CRD

### PR DESCRIPTION
**Pull request by bhatpmk**
This PR addresses the extended validation in Kubernetes v1.18.10, which causes the error mentioned in VZ-1747 during the installation of Verrazzano.

All other details provided in https://github.com/verrazzano/verrazzano-crd-generator/pull/17, including the result of acceptance test is applicable to this PR as well

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -85,12 +85,22 @@ build: go-mod
...
-	operator-sdk generate k8s
-	operator-sdk generate crds
+	$(CONTROLLER_GEN) object:headerFile=hack/boilerplate.go.txt paths=./pkg/...
+	$(CONTROLLER_GEN) crd:crdVersions=v1 output:crd:artifacts:config=deploy/crds paths=./pkg/...
+	mv deploy/crds/verrazzano.io_helidonapps.yaml deploy/crds/verrazzano.io_helidonapps_crd.yaml
 	./hack/add-crd-header.sh
 
+.PHONY: controller-gen
+controller-gen:
+	go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.1
```
Comment (markxnelson) [inline Makefile:97]: looks like you are introducing a new dependency here

File: THIRD_PARTY_LICENSES.txt [hunk 2]
```diff
@@ -1629,4 +1629,4 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 
 ATTRIBUTION-HELPER-GENERATED:
```
Comment (markxnelson) [inline THIRD_PARTY_LICENSES.txt:1631]: this looks odd - if yhou changed go.mod, we expect to see more change than just the md5sum

File: deploy/crds/verrazzano.io_helidonapps_crd.yaml [hunk 3]
```diff
@@ -3536,3 +3543,9 @@ spec:
     storage: true
     subresources:
       status: {}
+status:
```
Comment (markxnelson) [inline deploy/crds/verrazzano.io_helidonapps_crd.yaml:3546]: status should not be here - that is an output field

File: go.mod [hunk 4]
```diff
@@ -14,6 +14,7 @@ require (
 	k8s.io/gengo v0.0.0-20200114144118-36b2048a9120
 	k8s.io/kube-openapi v0.0.0-20200121204235-bf4fb3bd569c
 	sigs.k8s.io/controller-runtime v0.6.0
+	sigs.k8s.io/controller-tools v0.4.1 // indirect
```
Comment (markxnelson) [inline go.mod:17]: you can `go mod tidy` to get rid of indirects before you check in, in most cases

**Review by markxnelson — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (bhatpmk) [reply Makefile:98]: @markxnelson The module controller-gen is used to generate the CRD here. This change sets the default for protocol, and there by succeeds the extended validation introduced in k8s 1.18.10. Can you please approve this new dependency ?

File: THIRD_PARTY_LICENSES.txt (hunk 2 above)
Comment (bhatpmk) [reply THIRD_PARTY_LICENSES.txt:1631]: I have run the attribution-helper and this is the only change in the THIRD_PARTY_LICENSES.txt

File: go.mod (hunk 4 above)
Comment (bhatpmk) [reply go.mod]: Done

**Comment by bhatpmk** [comment]
I have a clean acceptance test using OKE Cluster v1.18.10 and the test source from a branch -
https://build.verrazzano.io/job/verrazzano-magic-dns-acceptance-tests/job/pabhat%252FVZ-1747/10/

**Review by mgianatagh — APPROVED** [review]
I tested the changes by doing a build, generate, and ran attribution-helper.  Everything worked fine.

**Review by bobdonat — DISMISSED** [review]

**Review by mgianatagh — APPROVED** [review]

**Comment by bhatpmk** [comment]
@markxnelson I have addressed the comments you have provided to this PR. Can you please take a look at the change ?

**Review by markxnelson — APPROVED** [review]

**Pull request merged**